### PR TITLE
Add CPT selector to Latest Posts

### PR DESCRIPTION
**Pull request by Ringish**
## Description

Adds a post type selector to the `LatestPosts` (in the `QueryControls`) block as described at #9013 and mentioned in #20046. 

## How has this been tested?

I've tested a few different scenarios where you would choose specific post types in the `LatestPosts` block.
## Screenshots 
Skärminspelning-2020-03-12-kl -14 23 38

## Types of changes

New feature 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/latest-posts/block.json [hunk 1]
```diff
@@ -1,72 +1,75 @@
...
-		"featuredImageSizeWidth": {
-			"type": "number",
-			"default":null
-		},
-		"featuredImageSizeHeight": {
-			"type": "number",
-			"default":null
-		}
-	}
+    "name": "core/latest-posts",
```
Comment (draganescu) [inline packages/block-library/src/latest-posts/block.json]: It looks like this file has been reformatted and for no good reason. We should revert this change to only include 

```
 "postType": {
   "type": "string"
 },
```

It's probably an editor spaces setting issue :)

File: packages/block-library/src/latest-posts/edit.js [hunk 2]
```diff
@@ -104,8 +118,9 @@ class LatestPostsEdit extends Component {
 			featuredImageSizeSlug,
 			featuredImageSizeWidth,
 			featuredImageSizeHeight,
+			postType,
 		} = attributes;
-		const suggestions = categoriesList.reduce(
+		const categorySuggestions = categoriesList.reduce(
```
Comment (draganescu) [inline packages/block-library/src/latest-posts/edit.js]: you changed this here from [my comment](https://github.com/WordPress/gutenberg/pull/20832#discussion_r391647710) in the other PR right?

File: packages/block-library/src/latest-posts/edit.js [hunk 3]
```diff
@@ -254,10 +274,11 @@ class LatestPostsEdit extends Component {
 									value: item.name || item.value,
 								} ) )
 							}
-							suggestions={ Object.keys( suggestions ) }
+							suggestions={ Object.keys( categorySuggestions ) }
```
Comment (draganescu) [inline packages/block-library/src/latest-posts/edit.js]: same, this is from the other PR. if we change in both places we might end up with conflicts 😁

**Review by draganescu — CHANGES_REQUESTED** [review]
Looks great. The only major change is that json file that got entirely commited and maybe we should rename that category related variable in the other PR.

Testing locally I noticed all post types are shown, including system types, which should be hidden:

Screenshot 2020-03-13 at 08 29 56

Normally from that list only Posts and Pages are thing one would want to filter on.

**Comment by draganescu** [comment]
Trac is also showing a linter issue [here](https://travis-ci.com/github/WordPress/gutenberg/jobs/297335656):

```
gutenberg/packages/block-library/src/latest-posts/edit.js
[0]   471:38  error  'getPostTypes' is assigned a value but never used  no-unused-vars
```

If you setup ESLint in VSCode these will be caught by the editor. Also Gutenberg has some git pre-commit hooks that catch these before the commit, I wonder why they don't run for you.

**Comment by ntsekouras** [comment]
Hey @Ringish. Thanks for all the work you've done so far in this PR! Do you think you'll find some time to wrap it up? It will be needed to enhance query related functionality.

**Comment by paaljoachim** [comment]
How is this PR coming along?

Order of controls should probably go like this.
User selects the Post Type.
Then category (or multiple) based on the Post Type selected above.
Then tags (if the user wants to add these in).
Order by
Number of posts to show (or all)

Screen Shot 2020-11-01 at 16 58 59

Screenshot from https://wordpress.org/plugins/flex-posts/

**Comment by ntsekouras** [comment]
Hey @paaljoachim - I think the focus has shifted towards making this block a `Query` block variation, which now has support for CPTs (the block has more work enqueued). Check this issue here: https://github.com/WordPress/gutenberg/issues/26193

**Comment by draganescu** [comment]
I would close this PR as it will be irrelevant even if it's finalized.

**Comment by paaljoachim** [comment]
Closing this PR based on above comments by Andrei @draganescu and Nik @ntsekouras 
The Add Posts List block will become a variation of the Query block. The issue for that is here: https://github.com/WordPress/gutenberg/issues/26193

**Pull request closed without merging**